aiida/daemon/runner.py

```diff
@@ -29,15 +30,28 @@ def start_daemon():
     configure_logging(daemon=True, daemon_log_file=daemon_client.daemon_log_file)
 
     runner = DaemonRunner(rmq_config=get_rmq_config(), rmq_submit=False)
-    set_runner(runner)
 
+    def shutdown_daemon(num, frame):
+        logger.info('Received signal to shut down the daemon runner')
+        runner.close()
```

Comment:
Following the tornado documentation, I think it should just be ``.stop`` in the callback, and ``.close()`` should be called still in the ``try`` block, after the ``start`` has returned.

Reply:
The `runner.close` method does not actually interact with the tornado event loop, it just calls `close` on the `RmqConnector` of the `Runner`.

Reply:
Looking at it, I think the loop is not closed now. That's also not necessary however, since the process terminates anyway and that will clean up everything.

Reply:
You are right, but that was exactly the intention. We just stop the loop and then the process exiting should automatically close the loop and clean everything up, according to the tornado docs link you sent, right?

Reply:
Yes, that's how I read it as well -- so it should be alright.